src/Node/ReadLine.purs

```diff
@@ -1,28 +1,74 @@
 -- | This module provides a binding to the Node `readline` API.
 
-module Node.ReadLine where
+module Node.ReadLine
+  ( Interface
+  , InterfaceOptions
+  , Completer
+  , LineHandler
+  , createInteractiveInterface
+  , createConsoleInterface
+  , prompt
+  , setPrompt
+  , setLineHandler
+  , close
+  , noCompletion
+  ) where
 
 import Prelude (return)
-
-import Control.Monad.Eff
-import Control.Monad.Eff.Console
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Node.Process (stdin, stdout)
+import Node.Stream (Readable, Writable)
 
 -- | A handle to a console interface.
 -- |
--- | A handle can be created with the `createInterface` function.
+-- | A handle can be created with the `createInteractiveInterface` function.
 foreign import data Interface :: *
 
+foreign import createIntetractiveInterfaceImpl :: forall opts eff. { | opts } -> Eff (console :: CONSOLE | eff) Interface
+
+foreign import undefined :: forall a. a
+
+-- | Options passed to `readline`'s `createInterface`
+type InterfaceOptions r w a e =
```

Comment:
Do you think console should appear in the effect rows here? If there can be situations where an interface has nothing to do with the console, eg when the input stream comes from, say, an HTTP request or a file, then perhaps it shouldn't?